cmake/system_check.cmake

```diff
@@ -31,7 +31,11 @@ endif()
 
 # Pretty thorough determination of arch. Add more if needed
 if(CMAKE_CL_64 OR MINGW64)
-  set(X86_64 1)
+  if (CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*|arm64.*|ARM64.*)")
```

Comment:
Won't this also require a change in line 27 to detect aarch64 so MINGW64 is set to 1?

Reply:
I mean when building with MSYS2's `clangarm64` environment